packages/engine/CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+[comment]: <> (This CONTRIBUTING guide was heavily inspired by the great one provided by the lovely folks at Atom [https://github.com/atom/atom/blob/master/CONTRIBUTING.md])
+
+# Contributing to hEngine
+
+Thank you for your interest in adding to the hEngine!
+
+We've provided the following guidelines to help you with contributions, in addition to the ones described below, we've
+also established a set of [community guidelines](https://hash.ai/legal/community) to enable as many people as possible
+to contribute to and benefit from HASH. Please follow these when interacting with this repo.
+
+#### Table Of Contents
+
+- [I have a question, where do I go?](#i-have-a-question--where-do-i-go-)
+
+- [Getting started, what do I need to know?](#getting-started--what-do-i-need-to-know-)
+    * [Design Decisions](#design-decisions)
+
+- [How Can I Contribute?](#how-can-i-contribute-)
+    * [Reporting Bugs](#reporting-bugs)
+        + [Before Submitting A Bug Report](#before-submitting-a-bug-report)
+        + [How Do I Submit A (Good) Bug Report?](#how-do-i-submit-a--good--bug-report-)
+    * [Suggesting Enhancements](#suggesting-enhancements)
+        + [Before Submitting An Enhancement Suggestion](#before-submitting-an-enhancement-suggestion)
+        + [How Do I Submit A (Good) Enhancement Suggestion?](#how-do-i-submit-a--good--enhancement-suggestion-)
+    * [Your First Code Contribution](#your-first-code-contribution)
+        + [Local development](#local-development)
+    * [Pull Requests](#pull-requests)
+
+- [Styleguides](#styleguides)
+    * [Rust Styleguide](#rust-styleguide)
+    * [JavaScript Styleguide](#javascript-styleguide)
+    * [Python Styleguide](#python-styleguide)
+    * [Documentation Styleguide](#documentation-styleguide)
+        + [Example](#example)
+
+- [Additional Notes](#additional-notes)
+    * [Issue and Pull Request Labels](#issue-and-pull-request-labels)
+
+## I have a question, where do I go?
+
+> **Note:** Please don't use GitHub issues for help, we have a friendly community on... TODO links to Discord, etc.
+
+* TODO
+
+## Getting started, what do I need to know?
+
+### Design Decisions
+
+> TODO
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+> TODO
+
+#### Before Submitting A Bug Report
+
+> TODO
+
+#### How Do I Submit A (Good) Bug Report?
+
+> TODO
```

Comment:
I guess my above comment could be addressed here. But I feel like a 'good' bug report also includes niceties like screenshots or (even better) a screen recording demonstrating the repro.

Reply:
Agreed, again I think we can take a fair bit of inspiration from the Atom contributing guide.
https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report

We of course don't need to split into two sections like they do. But I quite like how there is a concise clear set of instructions for the actual bug report, and then auxiliary recommendations for how to improve and enhance it.